bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABridgeHandler.java

```diff
@@ -0,0 +1,282 @@
+            }
+        }
+    }
+
+    private void run() {
+        try {
+            logger.trace("Executing refresh job");
+            refresh();
+            updateStatus(ONLINE);
+        } catch (PlugwiseHAInvalidHostException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_INVALID_HOSTNAME);
+        } catch (PlugwiseHAUnauthorizedException | PlugwiseHANotAuthorizedException e) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, STATUS_DESCRIPTION_INVALID_CREDENTIALS);
+        } catch (PlugwiseHACommunicationException e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_COMMUNICATION_ERROR);
+        } catch (PlugwiseHATimeoutException e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_TIMEOUT);
+        } catch (PlugwiseHAException e) {
+            logger.debug("Unhandled exception while refreshing the Plugwise Home Automation Controller {} - {}",
+                    getThing().getUID(), e.getMessage());
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void refresh() throws PlugwiseHAException {
+        if (this.getController() != null) {
+            logger.debug("Refreshing the Plugwise Home Automation Controller {}", getThing().getUID());
+            this.config = getConfig().as(PlugwiseHABridgeThingConfig.class);
+
+            PlugwiseHAController controller = this.getController();
+            if (controller != null) {
+                controller.refresh();
+            }
+
+            getThing().getThings().forEach((thing) -> {
+                ThingHandler thingHandler = thing.getHandler();
+                if (thingHandler instanceof PlugwiseHABaseHandler) {
+                    ((PlugwiseHABaseHandler<PlugwiseHAModel, PlugwiseHAThingConfig>) thingHandler).refresh();
+                }
+            });
+        }
+    }
+
+    private void cancelRefreshJob() {
+        synchronized (this) {
+            if (this.refreshJob != null) {
+                logger.debug("Cancelling refresh job");
+                this.refreshJob.cancel(true);
+                this.refreshJob = null;
+            }
+        }
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        // Only update bridge status if statusInfo has changed
+        ThingStatusInfo statusInfo = ThingStatusInfoBuilder.create(status, statusDetail).withDescription(description)
+                .build();
+        if (!statusInfo.equals(getThing().getStatusInfo())) {
+            if (status == ONLINE || (status == OFFLINE && statusDetail == COMMUNICATION_ERROR)) {
+                scheduleRefreshJob();
+            } else if (status == OFFLINE && statusDetail == CONFIGURATION_ERROR) {
+                cancelRefreshJob();
```

Comment:
I would assume if there is a configuration error the refresh job is never started. So why cancel it here?

Reply:
As now it is clear to me  the thing is reinitialized after a configuration change. (and dispose all ready calls `cancelRefreshJob()` it is no longer needed.